File: react_todo_frontend/src/domain/value-object/BodyVO.ts
//バリデーション
export const BodyVO = (body: string | null): string => {
    if (body === null || body.trim().length === 0 || body.length > 100) {
        throw new Error("本文は1文字以上100文字以下で入力してください");
    }
    return body; //エラーなし
}

File: go_todo_api/domain/vo/body_vo.go
package value_object

import (
	"errors"
	"strings"
)

type Body struct {
	value string
}

// Bodyの生成関数
func FromStringBody(v string) (Body, error) {
	trimmed := strings.TrimSpace(v)
	if len(trimmed) == 0 || len(trimmed) > 100 {
		return Body{}, errors.New("本文は1文字以上100文字以下である必要があります")
	}
	return Body{value: trimmed}, nil
}

// Bodyの値を取得するメソッド
func (b Body) Value() string {
	return b.value
}
